Sources/AWSLambdaRuntimeCore/LambdaHandler.swift

```diff
@@ -129,18 +129,42 @@ public protocol EventLoopLambdaHandler: ByteBufferLambdaHandler {
 public extension EventLoopLambdaHandler {
     /// Driver for `ByteBuffer` -> `In` decoding and `Out` -> `ByteBuffer` encoding
     func handle(context: Lambda.Context, event: ByteBuffer) -> EventLoopFuture<ByteBuffer?> {
-        switch self.decodeIn(buffer: event) {
+        let segment = context.tracer.beginSegment(name: "HandleEvent", baggage: context.baggage)
+        // TODO: record errors propagated in result types?
+        let decodedEvent = segment.subsegment(name: "DecodeIn") { _ in
+            self.decodeIn(buffer: event)
+        }
+        switch decodedEvent {
         case .failure(let error):
+            segment.addError(error)
+            segment.end()
             return context.eventLoop.makeFailedFuture(CodecError.requestDecoding(error))
         case .success(let `in`):
-            return self.handle(context: context, event: `in`).flatMapThrowing { out in
-                switch self.encodeOut(allocator: context.allocator, value: out) {
-                case .failure(let error):
-                    throw CodecError.responseEncoding(error)
-                case .success(let buffer):
-                    return buffer
+            // TODO: use NIO helpers?
```

Comment:
Yeah, let's kick off designing those soon, ticketified: [Provide NIO Futures TracingSupport extensions package (not sure how to call the module yet) #125](https://github.com/slashmo/gsoc-swift-tracing/issues/125)

Reply:
@tomerd packaging wise, I'm thinking we should separate out all such support packages from the swift-tracing repo... like `swift-tracing-support` and there we'd add all such wrappers, also for URLSession and similar...? WDYT about separating it entirely on a repo level (not just separate module), so those can be versioned separately -- I suspect there can be much more "churn" on the "sugar" APIs than on the core stable boring API after all.